### PR TITLE
Rename batch_quantity to batch_size

### PR DESCRIPTION
**Pull request by AdrianSosic**
This PR renames all `batch_quantity` method parameters to `batch_size`, which is more on point and easier to understand. Also, it adds a deprecation mechanism for `batch_quantity`.

### CODE REVIEW

**Comment by AdrianSosic** [comment]
@AVHopp, @Scienfitz, please note: this is really just a simple renaming PR, but you see a large diff because it's on top of the other campaign PR --> select the correct commit to compare against / use a proper git IDE / wait until the other PR is merged 🙃

File: baybe/telemetry.py
Comment (AdrianSosic) [inline baybe/telemetry.py:117]: @Scienfitz: This is the only part where I was not really sure what to do (i.e. see value, not key). Shall we rename that as well or rather keep as is?

Comment (Scienfitz) [reply baybe/telemetry.py:117]: hmm if we really change it then I guess theres no way around it to change this too, can merge these values in the analysis of telemetry data anyway

File: tests/test_deprecations.py
Comment (Scienfitz) [inline tests/test_deprecations.py:105]: in this situation I woudl actually expect the warning, not the error? its just a renaming and not moving to to some other place like with the flag

seems inconsistent with other renamings

**Review by Scienfitz — APPROVED** [review]
using error isntead of warning seems a bit inconsistent since its just a renaming, but I dont mind too much

**Review by AVHopp — APPROVED** [review]
I fully trust you in this one, since I guess it was just a search and replace and the potential aspects regarding the deprecation were already raised.

File: baybe/telemetry.py
Comment (AdrianSosic) [reply baybe/telemetry.py:117]: Alright then, changed as well

File: tests/test_deprecations.py
Comment (AdrianSosic) [reply tests/test_deprecations.py:105]: You see, in this particular case the problem is that there is no foolproof "autofix" that the deprecation mechanism could provide, which would be required if you just raise a warning (in which the code execution would go on and the deprecation mechanism would need to make the necessary modifications). If someone passed for example `batch_quantity=6` to the method, inside the function you'd see the two local variables `batch_quantity=6` AND `batch_size=5`. From that you could conclude that someone used the old interface. However, if for some weird reason (e.g. if the call happens programmatically of something) both arguments would be passed, the situation would be unclear.

I know, rather unlikely, but I think it's not too bad to just stick with the mantra "if autofixable --> raise warning; if something's ambiguous --> raise error". Anyway, it's juste temporary code ... 🙃

**Pull request merged**
